WalletWasabi.Gui/Shell/Commands/ToolCommands.cs

```diff
@@ -73,8 +71,10 @@ public ToolCommands(CommandIconService commandIconService, AvaloniaGlobalCompone
 
 		private void OnWalletManager()
 		{
-			var walletManagerViewModel = IoC.Get<IShell>().GetOrCreate<WalletManagerViewModel>();
-			if (Directory.Exists(Global.WalletsDir) && Directory.EnumerateFiles(Global.WalletsDir).Any())
+			var global = Locator.Current.GetService<Global>();
+
+			var walletManagerViewModel = IoC.Get<IShell>().GetOrCreateByType<WalletManagerViewModel>();
```

Comment:
Here we have the `Locator` and the `IoC`, they seems to have a very similar goal, what's the difference? Can we use the `IoC` only?

Reply:
Good point. Let's decide on this question outside the scope of this PR so I can merge it and not block Dan.

Reply:
The IoC call here (naming is poor I know) is composition rather than service location, its the only way to get stuff from the shell, its basically a simpler interface for MEF.

I think we can probably eliminate calls to it in future, so code looks more consistent.

So in short I think Locator wouldnt return an IShell and IoC would not find Global.